### ISSUE TITLE
Langfuse integration documentation

### ISSUE THREAD

**Issue by Xtreemrus** [Mirascope/mirascope#452]
### Description

Hello,

There's a section in the documentation dedicated to langfuse:
https://mirascope.io/docs/latest/integrations/langfuse/#streams

```python
from mirascope.core import openai, prompt_template
from mirascope.integrations.langfuse import with_langfuse


@with_langfuse()
@openai.call(
    model="gpt-4o-mini",
    stream=True,
    call_params={"stream_options": {"include_usage": True}},
)
@prompt_template("Recommend a {genre} book.")
def recommend_book(genre: str):
    ...


for chunk, _ in recommend_book("fantasy"):
    print(chunk.content, end="", flush=True)
```

With the current versions of langfuse and mirascope, it produces the following error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 16
      5 @with_langfuse()
      6 @openai.call(
      7     model="gpt-4o-mini",
   (...)
     11 @prompt_template("Recommend a {genre} book.")
     12 def recommend_book(genre: str):
     13     ...
---> 16 for chunk, _ in recommend_book("fantasy"):
     17     print(chunk.content, end="", flush=True)

TypeError: iter() returned non-iterator of type 'function'
```

This issue might be related to the langfuse library, as there were difficulties in properly logging the stream even with their native library.

### Python, Mirascope & OS Versions, related packages (not required)

```TOML
Windows 11
Python 3.11.7

openai: 1.42.0
mirascope: 1.0.3
langfuse: 2.44.1
```

**Comment by willbakst** [comment]
I'm not 100% what the problem is yet, but in the meantime you can use langfuse directly to trace streams like this:

```python
from collections.abc import Generator

from langfuse.decorators import langfuse_context, observe

from mirascope.core import openai, prompt_template


@openai.call("gpt-4o-mini", stream=True)
@prompt_template("Recommend a {genre} book")
def recommend_book(genre: str): ...


@observe(
    name="run_stream",
    as_type="generation",
    capture_input=False,
    capture_output=False,
)
def run_stream() -> Generator[str, None, None]:
    stream = recommend_book("fantasy")
    for chunk, _ in stream:
        yield chunk.content

    call_response = stream.construct_call_response()
    langfuse_context.update_current_observation(
        input=call_response.prompt_template,
        metadata=call_response.response,
        tags=list(call_response.metadata.get("tags", [])) or None,
        model=call_response.model,
        output=call_response.message_param.get("content", None),
        usage={
            "input": call_response.input_tokens,
            "output": call_response.output_tokens,
        },
    )


for content in run_stream():
    print(content, end="", flush=True)
```

not ideal, but should provide a stop-gap while I look further into this issue

**Comment by willbakst** [comment]
fix is now released in v1.0.5!

thanks for catching this :)
